internal/bminventory/inventory.go

```diff
@@ -748,6 +748,33 @@ func (b *bareMetalInventory) setBootstrapHost(ctx context.Context, cluster commo
 	return nil
 }
 
+func (b *bareMetalInventory) UpdateClusterInstallConfig(ctx context.Context, params installer.UpdateClusterInstallConfigParams) middleware.Responder {
+	log := logutil.FromContext(ctx, b.log)
+	var cluster common.Cluster
+	query := identity.AddUserFilter(ctx, "id = ?")
+
+	err := b.db.First(&cluster, query, params.ClusterID).Error
+	if err != nil {
+		log.WithError(err).Errorf("failed to find cluster %s", params.ClusterID)
+		if gorm.IsRecordNotFoundError(err) {
+			return installer.NewUpdateClusterInstallConfigNotFound().WithPayload(common.GenerateError(http.StatusNotFound, err))
+		} else {
+			return installer.NewUpdateClusterInstallConfigInternalServerError().WithPayload(common.GenerateError(http.StatusInternalServerError, err))
+		}
+	}
+
+	if err = json.Unmarshal([]byte(params.InstallConfigParams), &installcfg.InstallerConfigBaremetal{}); err != nil {
```

Comment:
i think you can just check that the string is json, no need to verify against our specific struct, since in any case there will be fields that we won;t be able to verify. Or if you still want to verify specific fields (like pullsecret etc') and can do it in the installcfg

Reply:
That's actually what I had as my first pass (testing for valid json), but I figured this was a pretty low-effort operation and it would guard against mistakes like type errors that would only then be detected when the install config is being generated later on (and might be harder to communicate to the user).

Why should we *not* do this here?

Reply:
in a week or two we ignition generation code in Golang will be pushed to assisted-service. For now it will run a a goroutine, and it will be responsible for everything related to ignition generation, including installcfg. So it is logical that ignition generation will know what to verify and how, and API will just call upon it. In addition, there was talk of maybe moving ignition generation and a bunch of other staff to a different service, only that service will know how to verify

Reply:
If the ignition generation code is going to run from the same image then that's even better, right? That means that the installcfg package is no longer used from two different images and there's no reason the api can't be tightly coupled with that type.

I mean I can remove this, but until we are sure we're going to move ignition generation out of assisted-service I don't think it makes much sense to provide a worse user experience. As I said before, it's incredibly easy for us to give this feedback to the user earlier and more directly, so why wouldn't we? Also it's easy to remove if and when we do want to push all of the installconfig logic into a separate service.

And either way, if we're really looking for the *best* way to do this I would argue that we should be using [the install config type defined by the openshift installer](https://github.com/openshift/installer/blob/b934cccb2d3c7d775e64b9b2388e13599977751d/pkg/types/installconfig.go#L61) directly rather than copying the structure over into our repo and trying to keep that up to date. If that were the case, both this repo and a hypothetical ignition generation service could both get the canonical definition of the install config from there.